eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -29,40 +84,26 @@ catch {
 foreach ($res in $responses)
 {
   if (!$res -or !$res.ref) {
-    LogDebug "No branch returned from the branch prefix $BranchPrefix on $Repo. Skipping..."
+    LogDebug "No branch returned from the branch prefix $CentralPRRegex on $Repo. Skipping..."
     continue
   }
   $branch = $res.ref
   $branchName = $branch.Replace("refs/heads/","")
-  try {
-    $head = "${RepoId}:${branchName}"
-    LogDebug "Operating on branch [ $branchName ]"
-    $pullRequests = Get-GitHubPullRequests -RepoId $RepoId -State "all" -Head $head -AuthToken $AuthToken
-  }
-  catch
-  {
-    LogError "Get-GitHubPullRequests failed with exception:`n$_"
-    exit 1
-  }
-
-  $openPullRequests = $pullRequests | ? { $_.State -eq "open" }
-  if ($openPullRequests.Count -gt 0)
-  {
-    LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has open pull Requests. Skipping"
-    LogDebug $openPullRequests.url
+  if (!($branchName -match $CentralPRRegex)) {
     continue
   }
-
-  LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has no associated open Pull Request. "
+  $hasCentralPROpened = $false
+  if ($CentralRepoId) {
+    $hasCentralPROpened = RetrievePRAndClose -pullRequestNumber $Matches["PrNumber"] 
```

Comment:
What does this function help with? I'm mostly asking because I think it is a pretty large function that is called only once and while reading the code it is kind of difficult to understand. I will say I missed the early return statements which made me misunderstand part of the logic here. I would suggest just putting it inline here.

Reply:
I personally don't like nested block more than two layers. Make me lose track. Can we put comments here instead?

Reply:
I don't think the function as it is actually helps with the nesting. In my opinion it makes the nesting worse given that we now have early return statements that cause parts to be completely skipped which is difficult to follow inline. 

Also I still don't think your code correctly handles the case where we don't have a centralrepoid but we do have an open PR against that branch. We don't want to delete those branches and the previous code would skip such branches.

Reply:
Redefine my logic. It is currently inline.